src/main/java/com/thirdcc/webapp/authorization/TransactionSecurityExpression.java

```diff
@@ -0,0 +1,31 @@
+package com.thirdcc.webapp.authorization;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class TransactionSecurityExpression {
+
+    private final ManagementTeamSecurityExpression managementTeamSecurityExpression;
+
+    public TransactionSecurityExpression(ManagementTeamSecurityExpression managementTeamSecurityExpression){
+        this.managementTeamSecurityExpression = managementTeamSecurityExpression;
+    }
+
+    /**
+     * Security expression to check on Transaction related endpoint:
+     *  <ul>
+     *      <li>This is created as a separate service to reduce marshalling event based preauthorise logic with management team preauthorise logic.</li>
+     *      <li>This is because ManagementTeamSecurityExpression does not concern about the eventId null or not, whereby Transaction related endpoint accepts null eventId under certain circumstances.</li>
+     *  </ul>
+     *
+     * @param eventId eventId of current transaction
+     * @return allow or disallow transaction related request
+     */
+    public boolean transactionAccess(Long eventId){
```

Comment:
any big concern to having this class for Authorization?
Look like putting in ManagementTeam is fine

_Just Curious_

Reply:
Just for better separation of concern
- ManagementTeamSecurityExpression -> validate solely based on management access right
- TransactionSecurityConcern -> validate based on user access right + eventId
  - Need to have different validation logic for null/not null event id

therefore I can see the transaction will slowly have more diverged validation logic compared to management team security expression

Reply:
👌